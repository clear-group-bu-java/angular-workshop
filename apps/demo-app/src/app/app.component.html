<mat-toolbar color="primary">
  <span>{{ title }}</span>

  <a mat-button [routerLink]="['/']"><mat-icon>home</mat-icon> Home</a>
  <a mat-button [routerLink]="['/addresses']">
    <mat-icon>location_on</mat-icon>
    Addresses
  </a>
  <a mat-button disabled>
    <mat-icon>person_search</mat-icon>
    People
  </a>
  <a mat-button [routerLink]="['/performance']">
    <mat-icon>speed</mat-icon>
    Performance Demo
  </a>
</mat-toolbar>

<div class="content">
  <router-outlet></router-outlet>

  <button [disabled]="isDenied$ | async">Click me</button>

  <pre>
    {{ people$ | async | json }}
  </pre>
</div>
