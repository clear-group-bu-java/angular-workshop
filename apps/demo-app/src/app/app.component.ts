import { PeopleService, PersonDto } from '@alten/apis/people';
import { Component } from '@angular/core';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.scss'],
})
export class AppComponent {
  public readonly title = 'Demo App';
  public name!: string;

  public people: Array<PersonDto> = [];

  constructor(private readonly peopleService: PeopleService) {
    this.peopleService.getPeople().subscribe((people) => {
      this.people = people;
    });
  }
}
