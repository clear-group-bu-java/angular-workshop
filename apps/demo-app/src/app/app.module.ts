import { CoreModule } from '@alten/core';
import { PeopleDataAddressModule } from '@alten/people/data/address';
import { PeopleDataPersonModule } from '@alten/people/data/person';
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { MatButtonModule } from '@angular/material/button';
import { MatIconModule } from '@angular/material/icon';
import { MatToolbarModule } from '@angular/material/toolbar';
import { PreloadAllModules, RouterModule, Routes } from '@angular/router';
import { ServiceWorkerModule } from '@angular/service-worker';

import { environment } from '../environments/environment';
import { AppComponent } from './app.component';
import { PerformanceDemoComponent } from './performance-demo/performance-demo.component';
import { PerformanceDemoModule } from './performance-demo/performance-demo.module';

const APP_ROUTES: Routes = [
  { path: '', redirectTo: '/home', pathMatch: 'full' },
  {
    path: 'home',
    loadChildren: () =>
      import('@alten/views/home').then((m) => m.ViewsHomeModule),
  },
  {
    path: 'addresses',
    loadChildren: () =>
      import('@alten/people/views/address').then(
        (m) => m.PeopleViewsAddressModule
      ),
  },
  { path: 'performance', component: PerformanceDemoComponent },
];
@NgModule({
  declarations: [AppComponent],
  imports: [
    CommonModule,
    RouterModule.forRoot(APP_ROUTES, {
      initialNavigation: 'enabled',
      relativeLinkResolution: 'corrected',
      preloadingStrategy: PreloadAllModules,
    }),

    ServiceWorkerModule.register('ngsw-worker.js', {
      enabled: environment.production,
    }),

    CoreModule.forRoot({
      environment,
    }),

    MatButtonModule,
    MatIconModule,
    MatToolbarModule,

    PeopleDataAddressModule,
    PeopleDataPersonModule,
    PerformanceDemoModule,
  ],
  providers: [],
  bootstrap: [AppComponent],
})
export class AppModule {}
