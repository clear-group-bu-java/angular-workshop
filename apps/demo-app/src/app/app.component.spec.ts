/* eslint-disable @typescript-eslint/no-explicit-any */
import { PeopleService } from '@alten/apis/people';
import { mock } from '@alten/core';
import { of } from 'rxjs';

import { AppComponent } from './app.component';

describe('AppComponent', () => {
  const peopleServiceMock = mock<PeopleService>();

  peopleServiceMock.getPeople.mockReturnValue(
    of(<any>[
      {
        id: '12345',
        firstName: 'Test',
        lastname: 'User',
        email: 'test@example.com',
        cv: '',
        addressId: '54321',
      },
    ])
  );

  let app: AppComponent;

  beforeEach(() => {
    app = new AppComponent(peopleServiceMock);
  });

  it('should create the app', () => {
    expect(app).toBeTruthy();
  });

  it('should load people', () => {
    expect(app.people).toMatchSnapshot();
  });
});
