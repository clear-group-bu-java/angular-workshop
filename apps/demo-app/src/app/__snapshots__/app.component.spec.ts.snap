// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should load people 1`] = `
Array [
  Object {
    "addressId": "54321",
    "cv": "",
    "email": "test@example.com",
    "firstName": "Test",
    "id": "12345",
    "lastname": "User",
  },
]
`;
